### ISSUE TITLE
Let user decide whether to call lr_scheduler.step() automatically in manual optimization mode.

### ISSUE THREAD

**Issue by shenmishajing** [PyTorchLightning/pytorch-lightning#12284]
## 🚀 Feature

Like `self.automatic_optimization` add a var name `automatic_lr_schedule` or `automatic_lr_schedule_when_manual_optimization` in LightningModule with default value False, which User can set it to True.

When it is set to False, everything will run as now, but when it is set to True and the module is running in manaul_optimization mode, pytorch lightning will automatically call lr_scheduler.step() as in automatic optimization mode.

### Motivation

See [here](https://pytorch-lightning.readthedocs.io/en/stable/common/optimizers.html#learning-rate-scheduling-manual).

> Before 1.3, Lightning automatically called lr_scheduler.step() in both automatic and manual optimization. From 1.3, lr_scheduler.step() is now for the user to call at arbitrary intervals.

In my use case(like train a GAN), I surely have to call `optimizer.zero_grad()`, `self.manual_backward()` and `optimizer.step()` by myself, but for lr_schedulers, what I want to do is just like things done by pytorch lightning in automatic_optimization mode.

Therefore, I want to add this flag to make pytorch lightning automatically call lr_scheduler.step() for my module instead of write the logical code by myself.

______________________________________________________________________

#### If you enjoy Lightning, check out our other projects! ⚡

- [**Metrics**](https://github.com/PyTorchLightning/metrics): Machine learning metrics for distributed, scalable PyTorch applications.

- [**Lite**](https://pytorch-lightning.readthedocs.io/en/latest/starter/lightning_lite.html): enables pure PyTorch users to scale their existing code on any kind of device while retaining full control over their own loops and optimization logic.

- [**Flash**](https://github.com/PyTorchLightning/lightning-flash): The fastest way to get a Lightning baseline! A collection of tasks for fast prototyping, baselining, fine-tuning, and solving problems with deep learning.

- [**Bolts**](https://github.com/PyTorchLightning/lightning-bolts): Pretrained SOTA Deep Learning models, callbacks, and more for research and production with PyTorch Lightning and PyTorch.

- [**Lightning Transformers**](https://github.com/PyTorchLightning/lightning-transformers): Flexible interface for high-performance research using SOTA Transformers leveraging Pytorch Lightning, Transformers, and Hydra.

cc @borda @rohitgr7 @akihironitta

**Comment by rohitgr7** [comment]
ideally, optimizer and scheduler are called in sequential order. In case if optimizer step isn't called by the user, we cannot guarantee whether to call the lr_scheduler step or not. So enabling such a case isn't reliable hence during manual optimization the complete access for the optimization process (of which lr_scheduler is a part of) is given to the user.

IMO, we shouldn't do this.

**Comment by shenmishajing** [comment]
Ok, you are right. Thans for your explanation. I will close this issue.
